sdk/test/Services/S3/IntegrationTests/TransferUtilityTests.cs

```diff
@@ -230,6 +230,55 @@ public void SimpleUploadWithSSE_C_LargeFile()
             UploadWithSSE_C(16 * MEG_SIZE, @"SimpleUploadTest\LargeFile");            
         }
 
+        [TestMethod]
+        [TestCategory("S3")]
+        public void DirectoryUploadDonwloadWithSSE_C()
+        {
+            var directoryTest = CreateTestDirectory();
+            var directoryTestPath = directoryTest.FullName;
+            var remoteDirectory = directoryTest.Name;
+
+            // Create an encryption key
+            Aes aesEncryption = Aes.Create();
+            aesEncryption.KeySize = 256;
+            aesEncryption.GenerateKey();
+            string base64Key = Convert.ToBase64String(aesEncryption.Key);
+
+            // Upload test directory with SSE-C
+            var transferUtility = new TransferUtility(Client);
+            var requestUpload = new TransferUtilityUploadDirectoryRequest
+            {
+                BucketName = bucketName,
+                Directory = directoryTestPath,
+                KeyPrefix = remoteDirectory,
+                SearchPattern = "*",
+                SearchOption = SearchOption.AllDirectories,
+
+                ServerSideEncryptionCustomerMethod = ServerSideEncryptionCustomerMethod.AES256,
+                ServerSideEncryptionCustomerProvidedKey = base64Key
+            };
+
+            transferUtility.UploadDirectory(requestUpload);
+
+            // Download remote test directory with SSE-C
+            var downloadPath = GenerateDirectoryPath();
+
+            var requestDownload = new TransferUtilityDownloadDirectoryRequest()
+            {
+                BucketName = bucketName,
+                S3Directory = remoteDirectory,
+                LocalDirectory = downloadPath,
+
+                ServerSideEncryptionCustomerMethod = ServerSideEncryptionCustomerMethod.AES256,
+                ServerSideEncryptionCustomerProvidedKey = base64Key
+            };
+
+            transferUtility.DownloadDirectory(requestDownload);
+
+            // Compare directories
+            UtilityMethods.CompareFiles(directoryTestPath, downloadPath);
```

Comment:
I ran this test locally and it failed, it looks like our `UtilityMethods.CompareFiles` method doesn't work when you pass a directory path (instead of a file path).

Changing that method is out of scope for this PR, so I think we could replace this section with something like this:
```
// Compare each file in both directories
var sourceFiles = Directory.EnumerateFiles(directoryTestPath, "*", SearchOption.AllDirectories).ToList();
var downloadedFiles = Directory.EnumerateFiles(downloadPath, "*", SearchOption.AllDirectories).ToList();

Assert.AreEqual(sourceFiles.Count, downloadedFiles.Count);
for (var i = 0; i < sourceFiles.Count(); i++)
{
    UtilityMethods.CompareFiles(sourceFiles[i], downloadedFiles[i]);
}
```

Reply:
Oh sorry, i miss understanding usage of UtilityMethods.CompareFiles.
Solution is soo big, i'm not able to open it or run it.
I update PR soon

Reply:
No problem, the `AWSSDK.Net*.sln` solutions include all services so they're slower to load. 

We do create separate solutions for each service, the S3 one is located at `.\sdk\src\Services\S3\S3.sln` (that's how I ran this specific test).